example/lib/main.dart

```diff
@@ -16,25 +16,57 @@ void main() {
 final RouteObserver<PageRoute> routeObserver = RouteObserver<PageRoute>();
 
 class MyApp extends StatelessWidget {
+  final GlobalKey<NavigatorState> _navigatorKey = GlobalKey();
+
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'FlutterBleLib example',
-      theme: new ThemeData(
-        primaryColor: new Color(0xFF0A3D91),
-        accentColor: new Color(0xFFCC0000),
+    return BlocProvider<NavigatorBloc>(
+      builder: (context) => NavigatorBloc(navigatorKey: _navigatorKey),
+      child: MaterialApp(
+        navigatorKey: _navigatorKey,
+        title: 'FlutterBleLib example',
+        theme: new ThemeData(
+          primaryColor: new Color(0xFF0A3D91),
+          accentColor: new Color(0xFFCC0000),
+        ),
+        initialRoute: PeripheralListScreen.routeName,
+        onGenerateRoute: (settings) {
+          switch (settings.name) {
+            case PeripheralListScreen.routeName:
+              {
+                return MaterialPageRoute(
+                  builder: (context) => BlocProvider(
+                    builder: (context) =>
+                        PeripheralListBloc(BleAdapterInjector.inject),
+                    child: PeripheralListScreen(),
+                  ),
+                );
+              }
+              break;
+            case PeripheralDetailsScreen.routeName:
+              {
+                return MaterialPageRoute(
```

Comment:
could you try to extract some method/factory/anything to improve readability? each case contains a lot of common code

Reply:
Introduced NavigatorRouteFactory class which takes responsibility for creating new routes.